src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/ResourceReferenceResolver.cs

```diff
@@ -70,10 +70,14 @@ public async Task ResolveReferencesAsync(Resource resource, IDictionary<string,
 
                         var results = await GetExistingResourceId(requestUrl, resourceType, conditionalQueries, cancellationToken);
 
-                        if (results == null || results.Count != 1)
+                        if (results == null)
                         {
                             throw new RequestNotValidException(string.Format(Core.Resources.InvalidConditionalReference, reference.Reference));
                         }
+                        else if (results.Count != 1)
```

Comment:
And could you change this check to be if the count is greater than 1.

Reply:
For conditional, we want exactly 1

Reply:
@LTA-Thinking , @brendankowitz Updated the if else statement which cover both count = 0 and count >1 condition.